src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapDirectoryIdentifier.cs

```diff
@@ -55,31 +55,7 @@ public LdapDirectoryIdentifier(string[] servers, int portNumber, bool fullyQuali
             PortNumber = portNumber;
         }
 
-        public string[] Servers
-        {
-            get
-            {
-                if (_servers == null)
-                {
-                    return Array.Empty<string>();
-                }
-                
-                var temporaryServers = new string[_servers.Length];
-                for (int i = 0; i < _servers.Length; i++)
-                {
-                    if (_servers[i] != null)
-                    {
-                        temporaryServers[i] = string.Copy(_servers[i]);
-                    }
-                    else
-                    {
-                        temporaryServers[i] = null;
-                    }
-                }
-
-                return temporaryServers;
-            }
-        }
```

Comment:
I assume you vetted the consumers of this and no one's doing anything wacky via mutation of the strings?

Reply:
Correct - the pattern appears to be from somebody who is more familiar with COM-style programming and incorrectly assumed that all elements in a managed collection similarly had to be duplicated or have a bumped ref count. I added reviewers who are more familiar with this code to confirm.